src/bson_mapper/archiver_mock.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <iostream>
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+
+#include <bson_mapper/bson_streambuf.hpp>
+
+// A mock class for testing serialization w/o an actual BSON serializer class.
+// This class is templated, but requires the given type to have primitive fields a, b, and c.
+// This will be removed once cereal::BSONArchiver is merged in
```

Comment:
Good point, I'll still change the comment(s) for clarity until we merge everything in.